### PR TITLE
Automatically generate dependabot changesets

### PR DESCRIPTION
**Pull request by jhaals**
## Hey, I just made a Pull Request!

This will hopefully save some recurring work by automating changeset generation in dependabot PRs.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: scripts/dependabot-changeset-maker.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+async function main() {
+  const branch = await runPlain('git', 'branch', '--show-current');
+  if (!branch.startsWith('dependabot/')) {
+    console.log('Not a dependabot branch');
+    return;
+  }
+
+  const diffFiles = await runPlain('git', 'diff', '--name-only', 'HEAD~1');
+  if (diffFiles.includes('.changeset')) {
+    console.log('Changeset already exists');
```
Comment (freben) [inline scripts/dependabot-changeset-maker.js:67]:
```suggestion
    console.log('Changeset already exists, skipping');
```

File: scripts/dependabot-changeset-maker.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    return;
+  }
+  const files = diffFiles
+    .split('\n')
+    .filter(file => file !== 'package.json') // skip root package.json
+    .filter(file => file.includes('package.json'));
+
+  if (!files.length) {
+    console.log('no package.json changes, skipping');
```
Comment (freben) [inline scripts/dependabot-changeset-maker.js:76]:
```suggestion
    console.log('No package.json changes, skipping');
```

**Review by freben — APPROVED** [review]
Looks good! Needs a prettier run on the workflow file tho

**Review by Rugvip — APPROVED** [review]
🎉 👌

**Comment by jhaals** [comment]
Will do this inside the github actions yaml for extra protection. Will be back.

**Pull request closed without merging**